pkg/webhook/admission/deployment/model_initializer_injector_test.go

```diff
@@ -594,3 +594,94 @@ func TestCredentialInjection(t *testing.T) {
 		g.Expect(c.Delete(context.TODO(), scenario.secret)).NotTo(gomega.HaveOccurred())
 	}
 }
+
+func TestModelInitializerConfigmap(t *testing.T) {
+	scenarios := map[string]struct {
+		original *appsv1.Deployment
+		expected *appsv1.Deployment
+	}{
+		"ModelInitializerConfig": {
+			original: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Annotations: map[string]string{
+								constants.ModelInitializerSourceUriInternalAnnotationKey: "gs://foo",
+							},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Name: "user-container",
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Annotations: map[string]string{
+								constants.ModelInitializerSourceUriInternalAnnotationKey: "gs://foo",
+							},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Name: "user-container",
+									VolumeMounts: []v1.VolumeMount{
+										{
+											Name:      "kfserving-provision-location",
+											MountPath: constants.DefaultModelLocalMountPath,
+											ReadOnly:  true,
+										},
+									},
+								},
+							},
+							InitContainers: []v1.Container{
+								{
+									Name:  "model-initializer",
+									Image: "model-initializer:v0.1",
+									Args:  []string{"gs://foo", constants.DefaultModelLocalMountPath},
+									VolumeMounts: []v1.VolumeMount{
+										{
+											Name:      "kfserving-provision-location",
+											MountPath: constants.DefaultModelLocalMountPath,
+										},
+									},
+								},
+							},
+							Volumes: []v1.Volume{
+								{
+									Name: "kfserving-provision-location",
+									VolumeSource: v1.VolumeSource{
+										EmptyDir: &v1.EmptyDirVolumeSource{},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for name, scenario := range scenarios {
+		injector := &ModelInitializerInjector{
+			credentialBuilder: credentials.NewCredentialBulder(c, &v1.ConfigMap{
+				Data: map[string]string{},
+			}),
+			config: &ModelInitializerConfig{
+				Image: "model-initializer:v0.1",
```

Comment:
Can make this :vFromConfig or something so it's explicitly from config?

Reply:
done!